### ISSUE TITLE
Feature request: enforcing consistent `nullable` property usage

### ISSUE THREAD

**Issue by mprisehk** [daniel7grant/eslint-plugin-typeorm-typescript#9]
### Problem
Specifying `{ nullable: false }` on a property is the same as leaving it off, since Columns are not nullable by default. I noticed that we use this option very inconsistently in our entity definitions. In the same entity, some fields are marked as `{ nullable: false }`, while others right next to it are not.
![image](https://github.com/daniel7grant/eslint-plugin-typeorm-typescript/assets/149178064/266329b0-a021-45fb-9266-4dd825ae5bf7)

This inconsistency slightly irks me because having a `Column({ nullable: false })` right next to a `Column()` seems to "softly" communicate that the second one is nullable.

### Solution options
1. Force all Columns to specify their nullable-ness. Either it is true or it is false.
2. Disallow mentioning `nullable: false` since it is the same as not specifying it. Any places that still specify `nullable` will be setting it to `true`, removing the confusion about the behavior of the unspecified `Column()` next to it.

### However
- Different people may prefer different behavior. Personally I prefer option 2, but I can see people preferring option 1, or the current situation which leaves all options open.

### Preferred solution
Because different people may prefer a different solution direction, maybe the ideal solution is introducing a new rule, which severity can be configured separately from the rest. It could have an [option](https://eslint.org/docs/latest/extend/custom-rules#accessing-options-passed-to-a-rule) to configure what the preferred behavior is. As an example:
```
//.eslintrc.json
"rules": {
    "typeorm-typescript/enforce-column-types": "error",
    "typeorm-typescript/enforce-relation-types": "error",
    // the new one:
    "typeorm-typescript/enforce-specifying-nullable": ["error", "only-non-default"], // options would be `always | only-non-default`
    // or:
    "typeorm-typescript/specify-default-nullable": ["error", "never"], // options would be `always | never`
}
```

### Remaining considerations
- For consistency it would also have to work with the relation types, which are nullable: true by default (hence the wording "only-non-default" and not "only-when-true")
- Is this something this plugin even aims to solve?
  - Up to you :) It is more of a style preference than a type error, but I think it makes sense as "an ESLint rule for TypeORM entities" and hence would be a good fit for this plugin.

**Comment by daniel7grant** [comment]
In my mind it is not that big of a problem, because the TypeScript type shows the correct nullability, and you will use that more throughout the codebase, but I understand where you are coming from.

First, I have to look at how difficult it would be to implement this, because we currently don't recommend any changes to the decorators (decorator is the "source of truth") so it would need some reshaping. But I believe it could be added to the  `typeorm-typescript/enforce-column-types` rule, I would imagine as an object parameter for clarity:

```json5
{
  "rules": {
    "typeorm-typescript/enforce-column-types": ["error", { "specifyNullable": "always" }],
    // or
    "typeorm-typescript/enforce-column-types": ["error", { "specifyNullable": "only-nullable" }],
    // or (default)
    "typeorm-typescript/enforce-column-types": ["error", { "specifyNullable": false }],
  }
}
```

Using this, I could add more conditions before the normal checks, so we could test that nullable is specified correctly.

**Comment by mprisehk** [comment]
Ah yeah I like that notation as well 👍 I understand if you decide not to implement this, but it has my vote :) Looking forward to see where you take it!

**Comment by daniel7grant** [comment]
I released `0.3.0`, and I decided it would be a cool feature so I added as a new rule [enforce-consistent-nullability](https://github.com/daniel7grant/eslint-plugin-typeorm-typescript/#typeorm-typescriptenforce-consistent-nullability) (you were correct, it was easier than patch the existing ones). Feel free to close, if you like this behaviour.

**Comment by mprisehk** [comment]
That's great! I tested it out and it works a treat. Thanks!
